components/scream/src/physics/shoc/shoc_main_impl.hpp

```diff
@@ -58,6 +63,33 @@ Int Functions<S,D>::shoc_init(
   const auto host_view = Kokkos::create_mirror_view(npbl_d);
   Kokkos::deep_copy(host_view, npbl_d);
 
+  // Allocate temporaries if using small kernels
+#ifndef SCREAM_MONOLITHIC_KERNELS
```

Comment:
If I understand correctly, this part is not following the established pattern for processes to allocate workspace. I believe this code should be in `SHOCMacrophysics::init_buffers` and be allocated using the `buffer_manager.get_memory()` pointer, as is done for `m_buffer`. The reason for this is then no process is holding onto a bunch of device memory that can't be used by any other process.

Reply:
@ambrad , thanks, I was wondering if I should use that (Buffer). I have pushed a commit that does this.